collector/prometheus_collector.go

```diff
@@ -55,6 +56,13 @@ func NewPrometheusCollector(collectorName string, configFile []byte, metricCount
 
 	minPollingFrequency := configInJSON.PollingFrequency
 
+	//If the exact URL was not specified, generate it based on the ip address of the container.
+	endpoint := configInJSON.Endpoint
+	if endpoint.URL == "" {
+		ipAddress := containerHandler.GetContainerIPAddress()
+		configInJSON.Endpoint.URL = endpoint.URLConfig.Protocol + "://" + ipAddress + ":" + endpoint.URLConfig.Port.String() + endpoint.URLConfig.Path
+	}
+
```

Comment:
Consider moving the configFile parsing to a shared method so you don't need this duplicate code.